### ISSUE TITLE
Do we need to review the alerts?

### ISSUE THREAD

**Issue by mgifford** [wet-boew/wet-boew#8886]
Looking at the different style of alerts here:
https://wet-boew.github.io/wet-boew-styleguide/design/alerts-en.html

Should this be revised?  I'd love to have some User Research done with screen reader users to determine if we should be adding ARIA alerts to this.  

Some of the information looks like it should be within an `<aside>` rather than a `<section>`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside

I'm really not satisfied with the accessibility of this example (surely there should be some semantics built into an alert):
```
<div class="alert alert-info">
   <p>...</p>
</div>
```

Looking at this:
https://www.canada.ca/en/services/benefits/ei/cerb-application.html

Do we need more options? 

Finally, how do we see consistent implementations of these patterns.

**Comment by duboisp** [comment]
> Where is the provisional alert pattern? Is there a link somewhere I can see it?

https://wet-boew.github.io/themes-dist/GCWeb/provisional-en.html#alert

**Comment by mgifford** [comment]
Ok, great! Do you know if there was any consideration of adding any ARIA to this? Maybe the Alert role:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role

This might be useful for Danger warnings.  It just needs testing.

**Comment by andrewnordlund** [comment]
The alert role would not be appropriate for the design pattern in the original post.

The alerts here: https://wet-boew.github.io/wet-boew-styleguide/design/alerts-en.html are generally static in nature.  `role=alert` is used to notify screen users of new information without having to move their focus.  It has the same effect as `aria-live="assertive" aria-atomic="true".

The MDN page: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role is a little confusing.  Near the top it says: 

> The alert role should only be used for **static** text content.

Then 4 paragraphs later:

> The most important thing to know about the alert role is that it's for **dynamic** content. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, **if a page loads with multiple visible alerts scattered throughout, no alert would be read as they were not dynamically triggered**.
(emphasis mine)

**Comment by Jefferydo** [comment]
Hi folks. Tests of WET Alerts with nifty ARIA semantics here: 
https://bati-itao.github.io/learning/coaches/assessment/AEM/alerts-static.html

So far we've tested new semantics for a static alert and three JavaScript approaches to triggering live regions.

There are mixed messages re. aria-roledescription in the specs. According to ARIA 1.3, the aria-roledescription is declared rather than the role. But according to APG examples, we're not to include the name of the role in the aria-roledescription. No reason's given, but presumably it's because it would cause an echo. I've assumed the ARIA spec is correct.

JAWS 2022.2202.38
NVDA 2021.3.5
macOS: VoiceOver 10 + Safari 16
iOS: VoiceOver 15 + Safari 15
TalkBack 13.1 + Android 11
Firefox 109
Chrome 109.0.5414.75
Edge 109.0.1518.61
